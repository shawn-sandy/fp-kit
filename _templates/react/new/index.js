/* eslint-disable camelcase */
module.exports = {
  prompt: ({ inquirer }) => {
    const questions = [
      {
        type: 'input',
        name: 'component_name',
        message: 'What is the component name?'
      },
      {
        type: 'input',
        name: 'dir',
        message: 'The name of the directory to store the component (optional)'
      }
    ]
    return inquirer
      .prompt(questions)
      .then(answers => {
        const { component_name, dir } = answers
        const path = dir ? `${dir}` : `${component_name.toLowerCase()}`
        const absPath = `components/react/${path}`
        return { ...answers, path, absPath }
      })
  }
}
