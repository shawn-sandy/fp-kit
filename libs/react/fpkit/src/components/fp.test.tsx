import React from 'react'
import { render, screen } from '@test/react'
import FP from '../components/fp'
import jest from 'jest-mock'
import { userEvent } from '@storybook/test'

describe('FP component', () => {
  it('renders a div by default', () => {
    const { container } = render(<FP />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders a span when specified', () => {
    const { container } = render(<FP as="span">Span</FP>)
    const span = container.querySelector('span')
    expect(span).toBeInTheDocument()
  })

  it('renders children', () => {
    const { container } = render(<FP>Hello, world!</FP>)
    expect(container.firstChild).toHaveTextContent('Hello, world!')
  })

  it('applies styles', () => {
    const style = { backgroundColor: 'red;' }
    const { container } = render(<FP styles={style}>Hello, world!</FP>)
    expect(container.firstChild).toHaveStyle(style)
  })

  it('does not render styles when renderStyles is false', () => {
    const { container } = render(
      <FP renderStyles={false} styles={{ backgroundColor: 'red' }}>
        Hello, world!
      </FP>,
    )
    expect(container.firstChild).not.toHaveStyle('background-color: red;')
  })

  it('passes through props', async () => {
    const handleClick = jest.fn()
    const { container } = render(
      <FP as="button" onClick={handleClick}>
        Hello, world!
      </FP>,
    )

    await userEvent.click(screen.getByRole('button'))
    expect(handleClick).toHaveBeenCalled()
  })

  it('applies ref', () => {
    const ref = React.createRef<HTMLDivElement>()
    render(<FP ref={ref}>Hello, world!</FP>)
    expect(ref.current).toBeTruthy()
  })
})
