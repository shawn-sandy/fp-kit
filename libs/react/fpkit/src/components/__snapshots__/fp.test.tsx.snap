// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FP component > renders a div by default 1`] = `<div />`;
